Cpp_test/lib/ipc-shm-receive.cpp

```diff
@@ -2,19 +2,29 @@
 
 IPCShmReceive::~IPCShmReceive()
 {
-
-	shm_unlink(this->info.method_name);
-	if (pthread_cond_destroy(&(this->shm_ptr->cond_re)) != 0)
-	{
-		std::cerr << "Error at pthread_cond_destroy(): receive" << std::endl;
-	}
-	if (pthread_cond_destroy(&(this->shm_ptr->cond)) != 0)
-	{
-		std::cerr << "Error at pthread_cond_destroy(): send" << std::endl;
-	}
-	if (pthread_mutex_destroy(&(this->shm_ptr->mutex)) != 0)
+	if (this->shm_ptr != nullptr)
 	{
-		std::cerr << "Error at pthread_mutex_destroy()" << std::endl;
+		int ret1 = pthread_cond_destroy(&(this->shm_ptr->cond));
+		if (ret1 != 0)
+		{
+			std::cerr << "Error at pthread_cond_destroy(): " << strerror(ret1) << std::endl;
+		}
+
+		pthread_mutex_trylock(&(this->shm_ptr->mutex));
+
+		unlock_mutex();
+
+		int ret3 = pthread_mutexattr_destroy(&(this->mutex_attr));
+		if (ret3 != 0)
+		{
+			std::cerr << "Error at pthread_mutexattr_destroy(): " << strerror(ret3) << std::endl;
+		}
+
+		int ret4 = pthread_mutex_destroy(&(this->shm_ptr->mutex));
```

Comment:
mutex_destroy should be before mutexattr_destroy,  just to be as closer as you can from unlock.